pallets/subtensor/src/epoch/run_epoch.rs

```
@@ -1433,12 +1440,20 @@ impl<T: Config> Pallet<T> {
         bonds: &[Vec<(u16, I32F32)>],   // previous epoch bonds
         consensus: &[I32F32],           // previous epoch consensus weights
     ) -> Vec<Vec<I32F32>> {
-        assert!(weights.len() == bonds.len());
+        let mut alphas = Vec::with_capacity(consensus.len());
+
+        if weights.len() != bonds.len() {
+            log::error!(
+                "compute_liquid_alpha_values: weights and bonds have different lengths: {:?} != {:?}",
+                weights.len(),
+                bonds.len()
+            );
```

This is nice, do you think it makes sense to add some logging where asserts have been removed to be able to detect other errors in the future where there is an array/vec length mismatch? Because the way it is now handled is by silently defaulting but could be hard to debug later if something goes wrong

---

Great idea, @l0r1s , thank you for the comment! I'll add the missing logging.